integration/hive/src/main/java/org/apache/carbondata/hive/MapredCarbonInputFormat.java

```diff
@@ -213,6 +223,45 @@ private QueryModel getQueryModel(Configuration configuration, String path)
         .build();
   }
 
+  private static CarbonTable getCarbonTable(Configuration tableProperties)
+      throws InvalidConfigurationException, IOException, SQLException {
+    String[] tableUniqueName = tableProperties.get("name").split("\\.");
+    String databaseName = tableUniqueName[0];
+    String tableName = tableUniqueName[1];
+    String tablePath = tableProperties.get(hive_metastoreConstants.META_TABLE_LOCATION);
+    String columns = tableProperties.get(hive_metastoreConstants.META_TABLE_COLUMNS);
+    String sortColumns = tableProperties.get("sort_columns");
+    String columnTypes = tableProperties.get(hive_metastoreConstants.META_TABLE_COLUMN_TYPES);
+    String partitionColumns =
+        tableProperties.get(hive_metastoreConstants.META_TABLE_PARTITION_COLUMNS);
+    String partitionColumnTypes =
+        tableProperties.get(hive_metastoreConstants.META_TABLE_PARTITION_COLUMN_TYPES);
+    if (partitionColumns != null) {
+      columns = columns + "," + partitionColumns;
+      columnTypes = columnTypes + ":" + partitionColumnTypes;
+    }
+    String[] columnTypeArray = HiveCarbonUtil.splitSchemaStringToArray(columnTypes);
+
+    AbsoluteTableIdentifier absoluteTableIdentifier = AbsoluteTableIdentifier
+        .from(tableProperties.get(hive_metastoreConstants.META_TABLE_LOCATION),
+            getDatabaseName(tableProperties), getTableName(tableProperties));
+    String schemaPath =
+        CarbonTablePath.getSchemaFilePath(absoluteTableIdentifier.getTablePath(), tableProperties);
+
+    CarbonTable carbonTable;
+    String carbonDataFile = CarbonUtil.getFilePathExternalFilePath(schemaPath, tableProperties);
+    if (carbonDataFile == null) {
```

Comment:
This method is called only when carbondataFile does not exist. Refer: https://github.com/apache/carbondata/pull/3878/files#diff-e4da0735e7ef96dca3a7bfedb5a8039dR96

Please remove this uncessary listing and directly use the CarbonTable.buildFromTableInfo() to build carbon table.

No need for inferSchema code also

Reply:
done.